fpgalib/dac.py

```diff
@@ -85,41 +85,43 @@
 
 # TODO: Integrate this with DAC class
 # TODO: Have dac_calibration_server not DACify SRAM data
-def dacify(wave_a, wave_b, trigger_idx=(16, 17)):
-    """Construct SRAM sequence for a list of waveforms.
+
+
+SRAM_MAX = 0x1FFF  # 8190 = 2**13 - 1
+
+
+def dacify(wave_a, wave_b, trigger_idx=None):
+    """Construct SRAM sequence for a list of waveforms (DACify waveforms).
 
     Warning: Lots of hardware details exposed here.
 
     Args:
-        wave_a (np.ndarray): Waveform for DAC A normalized to ???
-        wave_b (np.ndarray): Waveform for DAC B normalized to ???
+        wave_a (np.ndarray):
+        wave_b (np.ndarray): Waveform for DAC B. Maximum amplitude is 1, i.e.
+            values ranging from -1 to 1 indicate full scale.
 
-    Returns (np.ndarray): Array of SRAM data with shape ???
+    Returns:
+        (list): Each element is an integer representing a 32 bit SRAM word.
     """
-
     if not len(wave_a) == len(wave_b):
         raise Exception('Lengths of DAC A and DAC B waveforms must be equal.')
     for q in np.hstack((wave_a, wave_b)):
-        if q > 1.0:
+        if np.any(np.abs(q) > 1.0):
             raise Exception('Wave amplitude cannot exceed 1')
-    # Multiply wave by full scale of DAC. DAC is 14 bit 2's compliment
-    # so full scale is 13 bits, ie. 1 1111 1111 1111 = 1FFF
-    data_a = [long(np.floor(0x1FFF*y)) for y in wave_a]
-    data_b = [long(np.floor(0x1FFF*y)) for y in wave_b]
-    # Chop off everything except lowest 14 bits.
-    trunc_a = [y & 0x3FFF for y in data_a]
-    trunc_b = [y & 0x3FFF for y in data_b]
-    dac_a_dat = trunc_a
-    # Shift DAC B data by 14 bits.
-    dac_b_dat = [y << 14 for y in trunc_b]
+
+    # Multiply wave by full scale. Then truncate to lowest 14 bits. Then for dac
+    # B shift left by 14 bits.
+    dac_a_dat = [0x3FFF & (long(np.floor(SRAM_MAX*y))) for y in wave_a]
```

Comment:
Why do we have to truncate when we verified it doesn't exceed +/-1? Are you truncating the bits below 14 bit resolution? Should the comment say truncate to largest 14 bits?